pkgs/development/python-modules/tkinter/default.nix

```diff
@@ -1,34 +1,66 @@
 {
   lib,
-  stdenv,
   buildPythonPackage,
+  setuptools,
   python,
-  py,
-  isPyPy,
+  pythonOlder,
+  tcl,
+  tclPackages,
+  tk,
 }:
 
 buildPythonPackage {
   pname = "tkinter";
   version = python.version;
-  src = py;
-  format = "other";
-
-  # tkinter is included in PyPy, making this package a no-op.
-  installPhase = lib.optionalString (!isPyPy) (
-    ''
-      # Move the tkinter module
-      mkdir -p $out/${py.sitePackages}
-      mv lib/${py.libPrefix}/lib-dynload/_tkinter* $out/${py.sitePackages}/
-    ''
-    + lib.optionalString (!stdenv.hostPlatform.isDarwin) ''
-      # Update the rpath to point to python without x11Support
-      old_rpath=$(patchelf --print-rpath $out/${py.sitePackages}/_tkinter*)
-      new_rpath=$(sed "s#${py}#${python}#g" <<< "$old_rpath" )
-      patchelf --set-rpath $new_rpath $out/${py.sitePackages}/_tkinter*
-    ''
-  );
-
-  meta = py.meta // {
+  pyproject = true;
+
+  src = python.src;
+
+  prePatch = ''
+    mkdir $NIX_BUILD_TOP/tkinter
+
+    # copy the module bits and pieces from the python source
+    cp -v  Modules/{_tkinter.c,tkinter.h} ../tkinter/
+    cp -rv Modules/clinic ../tkinter/
+    cp -rv Lib/tkinter ../tkinter/
+
+    pushd $NIX_BUILD_TOP/tkinter
+
+    # install our custom pyproject.toml
+    cp ${./pyproject.toml} ./pyproject.toml
+    substituteInPlace pyproject.toml \
+      --replace-fail "@python_version@" "${python.version}" \
+      --replace-fail "@python_internal_dir@" "${python}/include/${python.libPrefix}/internal"
+
+  '' + lib.optionalString (pythonOlder "3.13") ''
+    substituteInPlace "tkinter/tix.py" --replace-fail \
+      "os.environ.get('TIX_LIBRARY')" \
+      "os.environ.get('TIX_LIBRARY') or '${tclPackages.tix}/lib'"
+  '';
+
+  build-system = [ setuptools ];
+
+  buildInputs = [
+    tcl
+    tk
+  ];
+
+  env = {
+    TCLTK_LIBS = toString [
+      "-L${lib.getLib tcl}/lib"
+      "-L${lib.getLib tk}/lib"
+      "-l${tcl.libPrefix}"
+      "-l${tk.libPrefix}"
+    ];
+    TCLTK_CFLAGS = toString [
+      "-I${lib.getDev tcl}/include"
+      "-I${lib.getDev tk}/include"
+    ];
+  };
```

Comment:
This behaves weirdly with various position attributes. Better to `inherit (python.meta) …;` (but as it’s pre-existing, not a blocker of course).